Cli/Command/CheckViolationsCommand.php

```diff
@@ -0,0 +1,155 @@
+        }
+
+        $output->writeln(sprintf('<info>Found GIT repository in <%s></info>', $gitRepository->getGitDir()));
+
+        if ($input->getOption('commits')) {
+            $revisionsList = new GitRevisionList($gitRepository, $input->getOption('commits'));
+            $diff = $gitRepository->getDiff($revisionsList);
+
+        } else {
+            $diff = $gitRepository->getHeadCommit()->getDiff();
+        }
+
+        if (!$diff) {
+            $output->writeln('<info>No diff found.</info>');
+
+            return;
+        }
+
+        $changedFiles = array();
+        foreach ($diff->getFiles() as $file) {
+            if ($file->getNewName()) {
+                if (!$input->getOption('ignore-assets') && $this->isAsset($file->getNewName())) {
+                    continue;
+                }
+
+                $changedFiles[] = $file->getNewName();
+            }
+        }
+
+        $output->writeln(sprintf('<info>Found [%d] changed files</info>', count($changedFiles)));
+        $output->writeln(sprintf('<info>Get violations from latest SensioLabs insight analysis...</info>', count($changedFiles)));
+
+        $sensioInsightAnalysis = $api->getProject($input->getArgument('project-uuid'))->getLastAnalysis();
+        $analysisLinkPage = sprintf('%s/projects/%s/analyses/%d', $api::ENDPOINT, $projectUuid, $sensioInsightAnalysis->getNumber());
+
+        $output->writeln(
+            sprintf('<info>Last analysis number is [%d], link page <%s></info>', $sensioInsightAnalysis->getNumber(), $analysisLinkPage)
+        );
+
+        if (!($sensioInsightAnalysis->getEndAt() instanceof \DateTime)) {
+            $output->writeln(
+                sprintf(
+                    '<bg=yellow;fg=red>The analysis number [%d], started at [%s] is not yet finished</bg=yellow;fg=red>',
+                    $sensioInsightAnalysis->getNumber(),
+                    $sensioInsightAnalysis->getBeginAt()->format('d D Y, H:i:s')
+                )
+            );
+
+            return;
+        }
+
+        if (!$sensioInsightAnalysis->getViolations()) {
+            $output->writeln(
+                sprintf('<info>No violations found in the last analysis [%d]</info>', $sensioInsightAnalysis->getNumber())
+            );
+
+            return;
+        }
+
+        $violationsByResources = array();
+        $globalViolations = array();
+        foreach ($sensioInsightAnalysis->getViolations() as $violation) {
+            if ('' == $violation->getResource()) {
+                $globalViolations[] = $violation->getMessage();
+                continue;
+            }
+
+            if (!$input->getOption('ignore-assets') && $this->isAsset($violation->getResource())) {
+                continue;
+            }
+
+            if (in_array($violation->getResource(), $changedFiles)) {
+                if (!isset($violationsByResources[$violation->getResource()])) {
+                    $violationsByResources[$violation->getResource()] = array();
```

Comment:
This can be removed. ``$violationsByResources[$violation->getResource()][] = $violation->getMessage();`` won't trigger notices even if the key does not exist yet (we are in a write context here, not a read one)